### PR TITLE
[xsnap] lockdown, harden, purify - first try

### PR DESCRIPTION
**Pull request by dckc**
@phoddie I'm retroactively turning this into a pull request.

I reset the `master` branch to 57dcfc0af914, the last commit where we were in sync.

[edit @kriskowal] This is part 2 of the retcon and I’ve pared out the codec regression.

### CODE REVIEW

File: xsnap/sources/xsnap-worker.c [hunk 1]
```diff
@@ -72,20 +71,6 @@ xsCallback gxSnapshotCallbacks[mxSnapshotCallbackCount] = {
...
-	xs_textdecoder_get_ignoreBOM, // 10
-	xs_textdecoder_get_fatal, // 11
-
-	xs_textencoder, // 12
-	xs_textencoder_encode, // 13
-	xs_textencoder_encodeInto, // 14
-
-	xs_base64_encode, // 15
-	xs_base64_decode, // 16
-
```
Comment (dckc) [inline xsnap/sources/xsnap-worker.c:88]: I don't think @phoddie meant to remove TextEncoder / TextDecoder ; it looks like some sort of version skew set in.

**Review by kriskowal — APPROVED** [review]

**Comment by dckc** [comment]
@warner have you checked that these are suitably well hidden by our SES shim? Should we test it? There's an observable difference: this draft of `lockdown()` freezes `globalThis` where the SES shim does not.

**Comment by kriskowal** [comment]
See https://github.com/agoric-labs/xsnap-pub/pull/13

**Comment by dckc** [comment]
> @warner have you checked that these are suitably well hidden by our SES shim?

Oh. This doesn't even add them to our `xsnap-worker.c`.

**Pull request closed without merging**